library/psa_crypto.c

```diff
@@ -1912,12 +1960,14 @@ psa_status_t psa_import_key( const psa_key_attributes_t *attributes,
         goto exit;
 
     /* In the case of a transparent key or an opaque key stored in local
-     * storage (thus not in the case of generating a key in a secure element
-     * or cryptoprocessor with storage), we have to allocate a buffer to
-     * hold the generated key material. */
+     * storage, we have to allocate a buffer to hold the generated key
```

Comment:
Honestly, the comment was a bit confusing.
The existing comment gives the impression that in the case of secure element or cryptoprocessor with storage, there is no local storage.
That's incorrect.
Maybe my corrected comment doesn't completely fix it, how about reworded to
 /* In the case of a transparent key or an opaque key stored in local storage , we have to allocate a buffer to hold the generated key material, In the case of  /* In the case of importing/generating a key in a secure element or cryptoprocessor with storage, the allocation in local storage is handled as part of key creation. */

Reply:
Actually that comment refers to dynamic secure element drivers, i.e. the old-style secure element interface (`MBEDTLS_PSA_CRYPTO_SE_C`). This would be a very useful thing to clarify now that there is another secure element interface.

For dynamic SE drivers, it is true that there's no dynamic allocation here if the secure element is one with storage (the dynamic SE interface makes a distinction between SE with and without storage, it isn't just two typical use cases like it is with the new driver interface).